lib/decimal.ex

```diff
@@ -1510,6 +1510,24 @@ defmodule Decimal do
     end
   end
 
+  @doc """
+  Returns the sum of all elements in the given enumerable.
+
+  ## Examples
+
+      iex> Decimal.sum([1, "2", Decimal.new("3")])
+      #Decimal<6>
+
+      iex> Decimal.sum([])
+      #Decimal<0>
+
+  """
+  doc_since("1.9.0")
+  @spec sum(Enumerable.t()) :: t()
+  def sum(enumerable) do
+    Enum.reduce(enumerable, Decimal.new(0), &Decimal.add/2)
```

Comment:
Does it need to explicitly be `Decimal.new(0)`? Couldn't it just be `0`? I had the idea that `Decimal.add/2` actually made the conversion implicitly 😊

Reply:
yup, that would work too. By doing the conversion manually first it's probably a little bit faster (since the catch-all clause in Decimal.add that calls Decimal.new is the last clause) but most likely doesn't make a big difference either way.